accounting/accounting_cli_functions.py

```diff
@@ -168,76 +168,45 @@ def add_job_records(dataframe):
     return job_records
 
 
-def view_jobs_run_by_username(conn, username, output_file):
-    # look up userid with username
-    userid = get_uid(username)
-    # get the information pertaining to a user in the jobs DB
-    select_stmt = (
-        "SELECT userid,id,t_submit,t_run,t_inactive,ranks,R FROM jobs where userid=?"
-    )
-    dataframe = pd.read_sql_query(select_stmt, conn, params=(userid,))
-    # if the length of dataframe is 0, that means
-    # the user specified was not found in the table
-    if len(dataframe.index) == 0:
-        return "User not found in jobs table"
-    else:
-        job_records = add_job_records(dataframe)
-        if output_file is None:
-            print_job_records(job_records)
+def view_job_records(conn, output_file, args):
+    # find out which args were passed and place them in a dict
+    params = {}
+    params_list = []
+
+    for key, val in vars(args).items():
+        if val != None and key != "path" and key != "output_file" and key != "func":
+            params[key] = val
+
+    select_stmt = ""
+    # generate the SELECT statement based on the parameters passed in
+    if "user" in params:
+        params["user"] = get_uid(params["user"])
+        params_list.append(params["user"])
+        select_stmt += "SELECT userid,id,t_submit,t_run,t_inactive,ranks,R FROM jobs WHERE userid=? "
```

Comment:
is the user allowed to not specify any of the options?  so they can output everything?  Perhaps could do:

```
select_stmt = "SELECT ....".
where_stmt = ""
```

and at the very end

`select_stmt += where_stmt`

and if where_stmt is empty string, no biggie?

Reply:
Oh, I see what you mean. That's probably a good idea. I didn't include returning all results as an option at first, but that update should be pushed now:

```
$ flux account -p path/to/db-file view-job-records
   UserID  Username         JobID         T_Submit            T_Run       T_Inactive  Nodes                                                                            R
0    2201     user1  530059362304 1600808153.99083 1600808154.01275 1600808154.09784      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
1    2201     user1  518164316160 1600808153.28221 1600808153.30560 1600808153.37464      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
2    2201     user1  504692211712 1600808152.47932 1600808152.50098 1600808152.57389      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
3    2201     user1  492377735168 1600808151.74527 1600808151.76970 1600808151.84127      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
4    2001  fluxuser  291655122944 1600808139.78101 1600808139.80519 1600808139.88820      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
5    2001  fluxuser  281706233856 1600808139.18806 1600808139.21133 1600808139.28879      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
6    2001  fluxuser  269895073792 1600808138.48468 1600808138.50479 1600808138.57930      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
7    2001  fluxuser  258989883392 1600808137.83465 1600808137.85604 1600808137.93684      1  {"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0"}}]}}
```